types/styled-components/test/index.tsx

```diff
@@ -1224,6 +1224,27 @@ function unionTest() {
     <StyledReadable kind="magazine" author="Hejlsberg" />; // $ExpectError
 }
 
+function unionPerformanceTest() {
```

Comment:
I don't think we need a dedicated test since if the existing perf metrics show a difference.

Let's try it without this test first and see if we see a difference. Otherwise let's separate test and fix into separate PRs (the fix based on the test once it's merged) so that we know the test is actually showing performance improvements.

Reply:
Sure, I removed the test for the time being.